anomdetect/src/test/java/com/expedia/adaptivealerting/anomdetect/CusumAnomalyDetectorTest.java

```diff
@@ -0,0 +1,111 @@
+import java.time.Instant;
+import java.util.List;
+import java.util.ListIterator;
+import static com.expedia.adaptivealerting.anomdetect.NSigmasClassifier.DEFAULT_STRONG_SIGMAS;
+import static com.expedia.adaptivealerting.anomdetect.NSigmasClassifier.DEFAULT_WEAK_SIGMAS;
+import static junit.framework.TestCase.assertEquals;
+
+/**
+ * @author kashah
+ */
+public class CusumAnomalyDetectorTest {
+    private static final double WEAK_SIGMAS = DEFAULT_WEAK_SIGMAS;
+    private static final double STRONG_SIGMAS = DEFAULT_STRONG_SIGMAS;
+    private static final double TOLERANCE = 0.01;
+    private static final int WARMUP_PERIOD = 25;
+
+    private static List<CusumTestRow> data;
+
+    @BeforeClass
+    public static void setUpClass() throws IOException {
+        readData_calInflow();
+    }
+
+    @Test
+    public void testDefaultConstructor() {
+        final CusumAnomalyDetector detector = new CusumAnomalyDetector();
+        assertEquals(WEAK_SIGMAS, detector.getWeakThresholdSigmas());
+        assertEquals(STRONG_SIGMAS, detector.getStrongThresholdSigmas());
+        assertEquals(WARMUP_PERIOD, detector.getWarmUpPeriod());
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testIllegalTailException() {
+        int tail = 4;
+        double observed = 4.0;
+        int warmUpPeriod = 0;
+        new CusumAnomalyDetector(tail, 10, warmUpPeriod, WEAK_SIGMAS, STRONG_SIGMAS, 0.16)
+                .classify(MetricPointUtil.metricPoint(Instant.now(), observed));
+    }
+
+    @Test
+    public void testEvaluate() {
+
+        final ListIterator<CusumTestRow> testRows = data.listIterator();
+        final CusumTestRow testRow0 = testRows.next();
+        final double observed0 = testRow0.getObserved();
+        final CusumAnomalyDetector detector = new CusumAnomalyDetector(1, observed0, WARMUP_PERIOD, WEAK_SIGMAS,
+                STRONG_SIGMAS, 0.16);
+        int noOfDataPoints = 1;
+
+        // Params
+        assertEquals(WEAK_SIGMAS, detector.getWeakThresholdSigmas());
+        assertEquals(STRONG_SIGMAS, detector.getStrongThresholdSigmas());
+        assertEquals(WARMUP_PERIOD, detector.getWarmUpPeriod());
+
+        while (testRows.hasNext()) {
+            final CusumTestRow testRow = testRows.next();
+            final double observed = testRow.getObserved();
+            AnomalyResult result = detector.classify(MetricPointUtil.metricPoint(Instant.now(), observed));
+
+            if (noOfDataPoints < WARMUP_PERIOD) {
+                assertEquals(AnomalyLevel.valueOf("UNKNOWN"), result.getAnomalyLevel());
+            } else {
+                assertApproxEqual(testRow.getSh(), detector.getSumHigh());
+                assertApproxEqual(testRow.getSl(), detector.getSumLow());
+                assertEquals(AnomalyLevel.valueOf(testRow.getLevel()), result.getAnomalyLevel());
+            }
+            noOfDataPoints += 1;
+        }
+    }
+
+    private static void readData_calInflow() {
```

Comment:
Method name should not be readData_calInflow(). "calInflow" refers to a specific dataset (traffic in and out of some building at UC Irvine).